src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```
@@ -100,11 +100,22 @@ private Uri GetOrCreateStorageAccountForBootDiagnostics()
 
             if (!string.IsNullOrEmpty(storageAccountName))
             {
-                var storageAccountResponse = storageClient.StorageAccounts.GetProperties(this.ResourceGroupName, storageAccountName);
-
-                if (!storageAccountResponse.StorageAccount.AccountType.Equals(AccountType.PremiumLRS))
+                try
                 {
-                    return storageAccountResponse.StorageAccount.PrimaryEndpoints.Blob;
+                    var storageAccountResponse = storageClient.StorageAccounts.GetProperties(this.ResourceGroupName,
+                        storageAccountName);
+                    if (!storageAccountResponse.StorageAccount.AccountType.Equals(AccountType.PremiumLRS))
+                    {
+                        return storageAccountResponse.StorageAccount.PrimaryEndpoints.Blob;
+                    }
+                }
+                catch (Exception e)
+                {
+                    if (!e.Message.Contains("ResourceNotFound"))
```

Exceptions are swallowed here. Need to throw it and/or output the error text.